pkg/daemon/certs/certs.go

```diff
@@ -106,69 +143,120 @@ func splitAtPEMCert() func(data []byte, atEOF bool) (advance int, token []byte,
 	}
 }
 
-// ensureCertsAreImported imports the expected certificates into the instance's root system trust store, if not already
-// present. Returns a boolean if any certificates were imported
-func ensureCertsAreImported(expectedCerts []*x509.Certificate) (bool, error) {
+// fileExists checks if the file at the given path exists. Returns true if it exists, false if not, and error otherwise.
+func fileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	} else if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else {
+		return false, fmt.Errorf("Failed to check if file %s exists: %w", path, err)
+	}
+}
+
+// copyFile copies the file contents from the source file to the destination. Creates the destination file if needed.
+func copyFile(src, dst string) error {
+	r, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Close(); err != nil {
+			klog.Errorf("Failed to close file %s", r.Name())
+		}
+	}()
+	w, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := w.Close(); err != nil {
+			klog.Errorf("Failed to close file %s", w.Name())
+		}
+	}()
+	_, err = w.ReadFrom(r)
```

Comment:
Isn't it better to use `io.Copy()` and allow it make the decision to use `ReadFrom()`?

Reply:
fair point